st2client/st2client/commands/action_alias.py

```diff
@@ -40,3 +49,98 @@ class ActionAliasGetCommand(resource.ContentPackResourceGetCommand):
+                                       'output. "all" will return all '
+                                       'attributes.'))
+        self.parser.add_argument('-w', '--width', nargs='+', type=int,
+                                 default=None,
+                                 help=('Set the width of columns in output.'))
+
+    @resource.add_auth_token_to_kwargs_from_cli
+    def run(self, args, **kwargs):
+        match, _ = self.manager.match(args.match_text, **kwargs)
+        return [match]
+
+    def run_and_print(self, args, **kwargs):
+        instances = self.run(args, **kwargs)
+        self.print_output(instances, table.MultiColumnTable,
+                          attributes=args.attr, widths=args.width,
+                          json=args.json, yaml=args.yaml)
+
+
+class ActionAliasExecuteCommand(resource.ResourceCommand):
+    display_attributes = ['name']
+
+    def __init__(self, resource, *args, **kwargs):
+        super(ActionAliasExecuteCommand, self).__init__(
+            resource, 'execute',
+            'Execute the command text by finding a matching ActionAlias.',
+            *args, **kwargs)
+
+        self.parser.add_argument('command_text',
+                                 metavar='command',
+                                 help=help)
+        self.parser.add_argument('-h', '--help',
+                                 action='store_true', dest='help',
+                                 help='Print usage for the given action.')
+        self.parser.add_argument('--trace-tag', '--trace_tag',
+                                 help='A trace tag string to track execution later.',
+                                 dest='trace_tag', required=False)
+        self.parser.add_argument('--trace-id',
+                                 help='Existing trace id for this execution.',
+                                 dest='trace_id', required=False)
+        self.parser.add_argument('-a', '--async',
+                                 action='store_true', dest='async',
+                                 help='Do not wait for action to finish.')
+        self.parser.add_argument('-u', '--user', type=str, default=None,
+                                 help='User under which to run the action (admins only).')
+
+        self.parser.add_argument('--attr', nargs='+',
+                                 default=self.display_attributes,
+                                 help=('List of attributes to include in the '
+                                       'output. "all" will return all '
+                                       'attributes.'))
+        self.parser.add_argument('-w', '--width', nargs='+', type=int,
+                                 default=None,
+                                 help=('Set the width of columns in output.'))
+
+    @resource.add_auth_token_to_kwargs_from_cli
+    def run(self, args, **kwargs):
+        action_alias, representation = self.manager.match(args.command_text, **kwargs)
```

Comment:
I'm fine with having this command for now, but for true server side matching and execution which would allow us to make clients event dumber I think we should also move this to the server side (e.g. "match and execute" endpoint or similar).

This is not a blocker right now though. We can leave it here and introduce this new API endpoint in the future. The CLI command is forward compatible anyway (once the feature is available on the server, we just need to change the code here).

This would also save client / user one API call which is actually good in this case, especially if you send every single line posted to chat to the API.